### ISSUE TITLE
`[extras]` logging on launch exposes DB credentials in URL

### ISSUE THREAD

**Issue by justanotherdot** [SergioBenitez/Rocket#1178]
Hello!

> The version of Rocket you're using. Ensure it's the latest, if possible.

```toml
rocket = "0.4.2"
rocket_contrib = { version = "0.4.2", features = ["diesel_postgres_pool"] }
```

> The operating system (distribution and version) where the issue occurs.

```
$ cat /etc/issue
Debian GNU/Linux 9 \n \l
```

> A brief description of the bug that includes:

On launch of a rocket application there is logging of configuration details. One of those details is the `[extras]` field, if set. The code [here](https://github.com/SergioBenitez/Rocket/blob/4e6a7ddd5fffe66185baade28b421b7e6a46c70d/core/lib/src/rocket.rs#L424-L428) and [here](https://github.com/SergioBenitez/Rocket/blob/4e6a7ddd5fffe66185baade28b421b7e6a46c70d/core/lib/src/logger.rs#L110) seems to show that it will be printed even on the `Critical` logging level.

When this logs it is exposing the URL for connecting to our database, but [by convention](https://www.postgresql.org/docs/10/libpq-connect.html) ("33.1.1.2. Connection URIs") this contains the password to the database. 

> How you uncovered the bug. Short, reproducible tests are especially useful.

1. Set `Config(...).build(env).extras("database", DATABASE_MAP)` prior to `launch`ing Rocket   
2. Run the application in production mode and look at logs on stdout or whatever sink is configured

> Ideas, if any, about what Rocket is doing incorrectly.

Rocket ought not to log anything out on launch for `[extras]` for production as it potentially contains sensitive information, or Rocket ought to provide a simple way to mark values for hiding.

**Comment by jebrosen** [comment]
This is a duplicate of #932 which was merged into #852. This may be worth a shorter-term fix, such as hiding the contents of "extras" based on the log level or under `#[cfg(not(debug_assertions))]`.

**Comment by justanotherdot** [comment]
Sorry for the dupe. Did try to have a fish for an existing issue but didn’t
manage to dig those ones up. Happy to subscribe to #852 and discuss any
workarounds there.
